### DISCUSSION TITLE
Questions regarding safety of Audio functions

### DISCUSSION THREAD

**Discussion by Kacper-Kondracki** [raysan5/raylib#4172] [Q&A]
Hello, I have question regarding safety of using sound functions without initializing AudioDevice first, or closing it while Sounds are still loaded.
1. Is using LoadSound before AudioDevice initialization is UB, or it fails without any repercussions?
2. Is using UnloadSound safe to be called on multiple times on the same Sound without UB?
3. Is using PlaySound, StopSound, PauseSound, ResumeSound safe to be called on sound without AudioDevice initialized without UB?
4. After successfully initializing AudioDevice and loading sounds, it is safe to try to use that loaded Sound after closing AudioDevice without UB?
5. Is closing AudioDevice, while there is sound playing UB?

**Comment by casperbear** [comment]
I only know that in RayLib 5.0 if you change your default audio device after launching your RayLib app, your RayLib app crashes. After the fix https://github.com/raysan5/raylib/issues/3743 it doesn't crash, but doesn't continue playing sounds either. (At least in my Windows 7 setup)

**Comment by Peter0x44** [comment]
Why do you need to do those things? They all sound like bad ideas.
